### ISSUE TITLE
sddm does not close extra session for 2nd user login (and uses lots of cpu)

### ISSUE THREAD

**Issue by kermitfrog** [sddm/sddm#1338]
steps to reproduce:
1. Log in to plasma
2. switch user
3. log in to plasma with different user
4. log out second user

expected behaviour:
system closes second sddm and X server, then switches back to X display where user 1 has their session

actual bahaviour:
sddm and second X server stay open and focused. when manually switching back to first user, sddm keeps open and uses ~50% CPU (see #323 )

System info:
Distro: Ubuntu 20.04 / KDE neon
Plasma: 5.20.3
KDE Frameworks: 5.76.0
Qt: 5.15.1
sddm: 0.18.1-2xneon+20.04+focal+build13

If not closing the second X server is actualy the expected behaviour, fixing #323 would seem more important

**Comment by mschabhuettl** [comment]
I'm expiriencing the same issue on Arch Linux. Switching user is basically not working unless I switch tty and force sddm to restart.

**Comment by mfraser** [comment]
Does this also cause problems if you log back in using the same user?

**Comment by kermitfrog** [comment]
> Does this also cause problems if you log back in using the same user?

If you mean, starting a second X session as the same user - sddm won't let me do that! Even if trying to start awesome, while already logged in using plasma, the greeter simply restarts itself. (Still works as different user).

On the bright side, the CPU usage droped to 1.25 %. That is still 1.25 % more than doing nothing should use - but at least it's an improvement.

EDIT:
current versions:
Plasma: 5.21.3
KDE Frameworks: 5.80.0
Qt: 5.15.2
Kernel Version: 5.8.0-43-generic
sddm: 0.19.0-2+20.04+focal+build24

**Comment by mfraser** [comment]
> > Does this also cause problems if you log back in using the same user?
> 
> If you mean, starting a second X session as the same user - sddm won't let me do that! Even if trying to start awesome, while already logged in using plasma, the greeter simply restarts itself. (Still works as different user).
> 
> On the bright side, the CPU usage droped to 1.25 %. That is still 1.25 % more than doing nothing should use - but at least it's an improvement.
> 
> EDIT:
> current versions:
> Plasma: 5.21.3
> KDE Frameworks: 5.80.0
> Qt: 5.15.2
> Kernel Version: 5.8.0-43-generic
> sddm: 0.19.0-2+20.04+focal+build24

No, log out as the current user and then try logging back in as the same user.

**Comment by kermitfrog** [comment]
> No, log out as the current user and then try logging back in as the same user.

Ok, tried that - no problem then (as expected). The greeter is replaced by the desktop session, so it can't use the CPU.

EDIT: just to make sure, you understand what I mean by switch user:
I do *not* mean logging out, and then logging back in as a different user.
I do mean using the "switch user" option to start a second session with a different user while my first session is still running. Then logging out as the second user and resuming my first session. In that case, the second sddm session (that was started by "switch user" is still running a greeter and causes the problem. By pressing `Ctrl-Alt-F2` I can switch to it.
